plugin/helpers.go

```diff
@@ -6,10 +6,24 @@ package plugin
 import "github.com/mattermost/mattermost-server/model"
 
 type Helpers interface {
-	// EnsureBot ether returns an existing bot user or creates a bot user with
+	// EnsureBot either returns an existing bot user or creates a bot user with
 	// the specifications of the passed bot.
 	// Returns the id of the bot created or existing.
 	EnsureBot(bot *model.Bot) (string, error)
+
+	// KVGetJSON retrievs a value based on the key.
+	KVGetJSON(key string, value interface{}) error
+
+	// KVSetJSON stores a key-value pair.
+	KVSetJSON(key string, value interface{}) error
+
+	// VKCompareAndSetJSON updates a key-value pair if the current
+	// value is equal to oldValue. Inserts a key-value pair if oldValue is nil.
+	// Will not update it the current value != oldValue.
```

Comment:
Sorry @zkry for bringing this up now, but I would recommend documenting the return values of this api here as it will help the person understand how to consume them at a single glance, otherwise they'll have to look at the implementation, hunt down docs for `p.API.KVCompareAndSet` and that's how they'll know what the return `bool` value means.
Something like:
```
	// Returns (false, err) if DB/marshal/unmarshal error occurred
	// Returns (false, nil) if current value != old value
	// Returns (true, nil) if current value == old value or new key is inserted
```
(copied from https://github.com/mattermost/mattermost-server/blob/master/plugin/api.go#L466)
Feel free to add/remove/update as per the api's need.

Reply:
Great, I'll add this documentation string. Thanks